### TESTE T PARA COMPARAR DOIS GRUPOS OU AMOSTRAS DEPENDENTES - TESTE T AMOSTRAS EMPARELHADAS - PAIRED T TESTE

Comparar a média entre dois grupos ou amostras dependentes. Amostras dependentes (emparelhadas) são selecionadas aleatoriamente de modo que os valores de observações dependam de valores de outras observações.

#### PREMISSAS:
- Premissa 1: Distribuição Normal. Esta é uma amostra grande? 
Não, se n < 30. Como o tamanho da amostra não é grande o suficiente (menos de 30), precisamos verificar se as diferenças dos pares seguem uma distribuição normal. Pode-se usar o o teste de Shapiro-Wilk em R **shapiro.test ()** para verificar se é uma distribuição normal. Observe que, se os dados não forem distribuidos normalmente, é recomendável usar um teste de duas amostras emparelhadas não paramétricas.

- Premissa 2: Amostras dependentes (emparelhadas)


#### HIPÓTESES:

- Ho: médias estatisticamente iguais
- Ha: médias estatisticamente diferentes

#### CRITÉRIOS:

- Se p-value for menor que nível de significância, rejeita-se Ho.

In [1]:
# DESATIVA NOTAÇÃO CIENTÍFICA
options(scipen = 999)

In [2]:
# DESATIVAR MENSAGENS DE WARNINGS (warn = -1) E ATIVAR (warn = 0)
options(warn = -1)

In [3]:
# CARREGAR BIBLIOTECAS
suppressPackageStartupMessages({
library(dplyr)
})

#### CARREGAR DATASET

Dataset com pesos médios de camundongos antes e depois de um certo tratamento.

#### TESTE:
Existe alguma diferença significativa nos pesos médios após o tratamento?

In [4]:
load ("paired_weight.Rdata")
dados_peso_medio <- as.data.frame(paired_weight)
str(paired_weight)

'data.frame':	20 obs. of  2 variables:
 $ group : Factor w/ 2 levels "after","before": 2 2 2 2 2 2 2 2 2 2 ...
 $ weight: num  200 191 193 213 241 ...


#### CÁLCULO DA DIFERENÇA

In [5]:
diferenca <- with(paired_weight,
                  weight[group == "before"] - weight[group == "after"])

#### TESTE DE NORMALIDADE

In [6]:
shapiro.test(diferenca)  


	Shapiro-Wilk normality test

data:  diferenca
W = 0.94536, p-value = 0.6141


As diferenças (diferença) não é significativamente diferente de uma distribuição normal. Portanto a hipótese alternativa é rejeitada. 

#### TESTE T

In [7]:
nivel_confianca <- 0.95
nivel_significancia <- 0.05

Teste_t3 <- t.test(weight ~ group, data = paired_weight, paired = TRUE)
Teste_t3


	Paired t-test

data:  weight by group
t = 20.883, df = 9, p-value = 0.0000000062
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 173.4219 215.5581
sample estimates:
mean of the differences 
                 194.49 


Resultados:
t Calculado: 20.8831
Intervalo de Confiança: [ 173.4219 ; 215.5581 ]
t tabelado: [ -2.26 ; 2.26 ]

Conforme o teste, percebe-se que a verdadeira diferença entre as medias não é igual a 0. Logo rejeita-se Ho, pois 
as médias são diferentes estatisticamente, com 95 % de confiança.

#### TESTE T - FÓRMULA

In [ ]:
nivel_confianca <- 0.95
nivel_significancia <- 0.05

t_tabelado = round(qt(1-(nivel_significancia/2), df = Teste_t3$parameter),2)

cat("Resultados:")
cat("\nt Calculado:", round(Teste_t3$statistic, 4))
cat("\nIntervalo de Confiança: [", round(Teste_t3$conf.int[1], 8),";", round(Teste_t3$conf.int[2], 8),"]")
cat("\nt tabelado: [", -t_tabelado,";", t_tabelado,"]")

if ((Teste_t3$statistic >= (-t_tabelado)) & (Teste_t3$statistic <= t_tabelado)){
    cat("\n\nConforme o teste, percebe-se que a verdadeira diferença entre as médias é igual a 0. Logo não rejeita-se Ho, 
pois as médias são iguais estatisticamente, com",nivel_confianca*100,"% de confiança.")
} else{
    cat("\n\nConforme o teste, percebe-se que a verdadeira diferença entre as medias não é igual a 0. Logo rejeita-se Ho, pois 
as médias são diferentes estatisticamente, com",nivel_confianca*100,"% de confiança.")
    
}

Com base no teste, conclui-se que o peso médio dos camundongos antes do tratamento é significativamente diferente do peso médio após o tratamento com um p-value de 0.0000000062.